sensor: mpu6050: switch to devicetree and provide sample

pabigot opened the pull request:
Converts the driver to use devicetree properties instead of Kconfig.

Adds a sample demonstrating on-demand and triggered sampling.

NB: I find that the driver returns `-EIO` on sample fetch within the first hundred observations.   It seems likely some timing constraint is being violated.

pabigot committed:
drivers: sensor: mpu6050: convert to devicetree

Add a binding for the sensor and replace all Kconfig selection of
hardware parameters with devicetree property values.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

pabigot committed:
samples: sensor: mpu6050: convert to devicetree

Add a sample for the MPU6050 that demonstrates on-demand and triggered
display of all sensor data.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

dbkinder commented on samples/sensor/mpu6050/README.rst:
```diff
@@ -0,0 +1,59 @@
+Description
+***********
+
+This sample application periodically (0.5 Hz) measures samples the
```
Use either ``samples`` or ``measures``, but not both words.  Since you say "samples" below, I'd change this then to say just ``samples`` too.

pabigot replied:
I went with measures/measurements to avoid confusion with the fact this is a sample application.

dbkinder reviewed (dismissed):
one little tweak...

MaureenHelm commented on samples/sensor/mpu6050/src/main.c:
```diff
@@ -0,0 +1,120 @@
+	}
+	if (rc == 0) {
+		printf("[%s]:%g Cel\n"
+		       "  accel % f % f % f g\n"
```
accel units should be m/s/s

pabigot committed:
drivers: sensor: ak8975: remove legacy reference to MPU6050

MPU6050 no longer using Kconfig to specify I2C addresses.  Reference
to the removed symbol causes QA diagnostics.  Remove the reference;
when AK8975 is converted to devicetree it may be possible to restore
the link between the sensors.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

pabigot commented:
@MaureenHelm Units fixed.  I added a patch to the ak8975 internal header to eliminate the Kconfig warning, but that device needs to be converted to devicetree.

MaureenHelm commented on samples/sensor/mpu6050/boards/nrf52_pca10040.overlay:
```diff
@@ -0,0 +1,15 @@
+		reg = <0x68>;
+		status = "okay";
+		label = "MPU6050";
+		int-gpios = <&gpio0 11 0>;
```
missing flags

pabigot replied:
Actually not missing here: this is targeting master, and `GPIO_ACTIVE_HIGH` isn't available yet.

MaureenHelm replied:
Right, thanks.

MaureenHelm reviewed (changes requested):
> Units fixed. I added a patch to the ak8975 internal header to eliminate the Kconfig warning, but that device needs to be converted to devicetree.

Thanks.
